src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsSdk/DynamoDbRequestHandler.cs

```diff
@@ -23,7 +23,7 @@ public static AfterWrappedMethodDelegate HandleDynamoDbRequest(InstrumentedMetho
 
             // PutItemRequest => put_item,
             // CreateTableRequest => create_table, etc.
-            operation = GetOperationNameFromRequestType(requestType);
+            operation = _operationNameCache.GetOrAdd(requestType, GetOperationNameFromRequestType(requestType));
```

Comment:
When calling `GetOrAdd` in this manner, it will also call `GetOperationNameFromRequestType` every time. In order to prevent the `GetOperationNameFromRequestType` method from being invoked every time, you need to pass a reference to that method. Something like the following should work, because the `requestType` is also the cache key, and the key is passed to the method when an entry needs to be added to the cache.
```suggestion
            operation = _operationNameCache.GetOrAdd(requestType, GetOperationNameFromRequestType);
```